tests/test_variance_plane.py

```diff
@@ -0,0 +1,917 @@
+# This file is part of meas_algorithms.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import re
+import unittest
+from contextlib import nullcontext
+
+import galsim
+import lsst.utils.tests
+import matplotlib.colors as mcolors
+import matplotlib.pyplot as plt
+import numpy as np
+from lsst.ip.isr.isrMock import IsrMock
+from lsst.meas.algorithms import remove_signal_from_variance
+from lsst.utils.tests import methodParametersProduct
+from matplotlib.legend_handler import HandlerTuple
+from matplotlib.patheffects import withStroke
+from matplotlib.ticker import FixedLocator, FuncFormatter
+
+# Set to True to save the plot of the variance plane before and after
+# correction for a representative test case.
+SAVE_PLOT = True
+
```

Comment:
I've left the plotting parts in for now, knowing we'll likely remove them before finalizing. I thought they might be useful for the review.

Reply:
Just set it to `False` before merging. It's useful diagnostic to have when things go wrong but not worth saving a copy for every run.